src/Components/App.js

```diff
@@ -150,122 +148,77 @@ class App extends Component {
   }
 
   setFilter = (event) => {
-    let dataId = event.target.dataset.id;
-    let holder = { ...this.state.filterOptions };
-    let inputValue = document.querySelector('#input-value').value
+    const dataCategory = event.target.dataset.category;
+    const dataFilter = event.target.dataset.filter;
+    let filterOptions = { ...this.state.filterOptions };
+    const inputValue = document.querySelector('#input-value').value
     this.closePopUp();
 
-    switch (dataId) {
-
-      case 'card':
-        if (document.querySelector('#card').checked) {
-          holder.type.card = true;
-          holder.type.board = false;
-        } else {
-          holder.type.card = false
-        }
-        document.querySelector('#board').checked = false
-        break;
-      case 'board':
-        if (document.querySelector('#board').checked) {
-          holder.type.board = true;
-          holder.type.card = false;
-        } else {
-          holder.type.board = false
-        }
-        document.querySelector('#card').checked = false
-        break;
-
-      case 'two':
-        (document.querySelector('#two').checked) ? holder.players.two = true : holder.players.two = false;
-        break;
-      case 'threefour':
-        (document.querySelector('#threefour').checked) ? holder.players.threefour = true : holder.players.threefour = false;
-        break;
-      case 'fivesix':
-        (document.querySelector('#fivesix').checked) ? holder.players.fivesix = true : holder.players.fivesix = false;
-        break;
-      case 'seven':
-        (document.querySelector('#seven').checked) ? holder.players.seven = true : holder.players.seven = false;
-        break;
-
-      case 'lesseight':
-        (document.querySelector('#lesseight').checked) ? holder.age.lesseight = true : holder.age.lesseight = false;
-        break;
-      case 'eightthirteen':
-        (document.querySelector('#eightthirteen').checked) ? holder.age.eightthirteen = true : holder.age.eightthirteen = false;
-        break;
-      case 'thirteenplus':
-        (document.querySelector('#thirteenplus').checked) ? holder.age.thirteenplus = true : holder.age.thirteenplus = false;
-        break;
-      case 'adult':
-        (document.querySelector('#adult').checked) ? holder.age.adult = true : holder.age.adult = false;
-        break;
+    if (dataCategory && document.querySelector(`#${dataFilter}`).checked) {
+      filterOptions[dataCategory][dataFilter] = true
+    } else if (dataCategory) {
+      filterOptions[dataCategory][dataFilter] = false;
+    }
```

Comment:
This is the if statement that takes care of all the identical switch statements

Reply:
This looks SO good! So much cleaner, nice work. I'm glad we're using a more dynamic approach now.